woocommerce/woocommerce-square
# Update the Apple Developer domain verification file

### All Submissions:


---

### Changes proposed in this Pull Request:

As reported in #122, new sites are running into issues with getting their domain verified with Apple. It seems the domain verification file we bundle with this extension needs to be updated so this PR handles that.

Note that for any sites that are already verified, it doesn't seem like this impacts them, as things still work. But new sites can't be verified without this change.

Also note that Square is recommending that we don't store this file for longer than 24 hours as it can change in the future. Not sure how we go about changing that but something we may want to consider going forward.

Just to capture how things work right now:

1. When the Square plugin is active and we are in the admin on either the Square settings page or Square Gateway settings page, we fire off a method to check the Apple Pay domain registration (`Digital_Wallet\apple_pay_domain_registration`)
2. This method checks if the gateway is enabled and if digital wallets are enabled, as well as if Apple Pay is turned on. If all match, it then runs a method to check for the existence of the verification file (`Digital_Wallet\check_apple_pay_verification_file`)
3. This method will check if the domain verification file exists on the server and if it doesn't, it will copy the version from the plugin to the server root within the `.well-known` directory. It also checks if the contents of the existing file match what is in the plugin and if not, it will copy over the new file (important for us here, as this means the new file in this PR will be copied over even for sites that already have this file in place)
4. If we have a domain verification file in place and we aren't already registered, we then run a method to register the domain with Apple (`Digital_Wallet\register_apple_pay_domain`)
5. This method makes a remote request to `https://connect.squareup(sandbox).com/v2/apple-pay/domains` to verify the domain. If something goes wrong, we log an error message but otherwise we consider the domain verified

Closes #122 

### Steps to test the changes in this Pull Request:

1. Create a new, publicly accessible environment and install and configure WooCommerce
2. Checkout this PR and build the Square extension (`npm install && npm run build`)
3. Install and configure the Square extension. Turn on the Square payment gateway and ensure Digital Wallets are turned on. Ensure your domain is verified on the Square side (see https://woocommerce.com/document/woocommerce-square/payment-settings/#section-6)
4. Ensure no error messages are shown around domain verification
5. Using Safari, ensure the Apple Pay buttons show as expected. Note there's an existing bug where Apple Pay doesn't show in block-based carts or checkouts so will need to test with classic integrations 

### Changelog entry

> Fix - Update the Apple Pay domain verification file.
